File: app/src/main/java/space/pal/sig/model/dto/NewsDto.java
package space.pal.sig.model.dto;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class NewsDto {

    @SerializedName("name")
    private String name;
    @SerializedName("news_id")
    private String newsId;
    @SerializedName("url")
    private String url;
    @SerializedName("publication")
    private String publication;
    @SerializedName("mission")
    private String mission;
    @SerializedName("abstract")
    private String abstractText;
    @SerializedName("credits")
    private String credits;
    @SerializedName("thumbnail")
    private String thumbnail;
    @SerializedName("thumbnail_retina")
    private String thumbnailRetina;
    @SerializedName("thumbnail_1x")
    private String thumbnail1x;
    @SerializedName("thumbnail_2x")
    private String thumbnail2x;
    @SerializedName("keystone_image_1x")
    private String keystoneImage1x;
    @SerializedName("keystone_image_2x")
    private String keystoneImage2x;
    @SerializedName("release_images")
    private List<Integer> releaseImages;

}

File: app/src/main/java/space/pal/sig/model/dto/NewsDto.kt
package space.pal.sig.model.dto

import com.google.gson.annotations.SerializedName
import space.pal.sig.model.entity.News
import space.pal.sig.model.entity.NewsSource
import java.util.*

/**
 * SpaceNow
 * Created by Catalin on 12/25/2020
 **/
data class NewsDto(
        @SerializedName("name", alternate = ["title"])
        val title: String,
        @SerializedName("description", alternate = ["abstract"])
        val description: String,
        @SerializedName("pub_date", alternate = ["publication"])
        val date: Date,
        @SerializedName("link", alternate = ["url"])
        val link: String,
        @SerializedName("image", alternate = ["keystone_image_2x"])
        val image: String,
        @SerializedName("thumbnail")
        val thumbnail: String
) {

    fun toNews(source: NewsSource): News {
        val url = link.replace("http://", "https://")
        return News(
                title, description, date, url, image, thumbnail, source
        )
    }

}
